go/payments/cmd/calc_payments/main.go

```diff
@@ -170,110 +178,116 @@ func calcCellwisePayments(ctx context.Context, dEnv *env.DoltEnv, opts options)
 		defer stopProfFunc()
 	}
 
-	i, att, err := readIncremental(opts.startHash, mergeCommits, opts.incremantalDir)
-
+	i, summary, err := readLatestSummary(ctx, method, mergeCommits)
 	if err != nil {
 		return err
 	}
 
-	for ; i >= 0; i-- {
-		currHash, err := mergeCommits[i].HashOf()
+	var prevCommit *doltdb.Commit
+	prevCommitIdx := len(mergeCommits) - i - 1
+	if prevCommitIdx >= 0 {
+		prevCommit = mergeCommits[len(mergeCommits)-prevCommitIdx-1]
+	}
+
+	if summary == nil {
+		if prevCommitIdx >= 0 {
+			return errors.New("failed to read summary for latest commit")
+		}
+
+		summary = method.EmptySummary(ctx)
+	}
 
+	for i--; i >= 0; i-- {
```

Comment:
yeah.  And one really nasty bug I had came out of this.  Gonna refactor this.  Will start by reversing the merge commit order so that value 0 is the oldest.

Reply:
After that should be able to make this easier to read